### PR TITLE
Improve/add some French locales

### PR DESCRIPTION
**Pull request by eramdam**
I saw that Mastodon was lacking some French strings so I took time to write them down and/or fix some already existing ones.
**However** I did not run Mastodon locally with those new locales to see if the UI breaks or not, I will try to do that later but if someone wants to do it, that would be awesome too!

### CODE REVIEW

File: config/locales/simple_form.fr.yml [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
         password: Mot de passe
+        setting_default_privacy: Confidentialité des statuts
+        type: Type d'import
         username: Identifiant
       interactions:
         must_be_follower: Masquer les notifications des personnes qui ne vous suivent pas
         must_be_following: Masquer les notifications des personnes que vous ne suivez pas
       notification_emails:
+        digest: Envoyer des emails récapitulatifs
         favourite: Envoyer un courriel lorsque quelqu’un ajoute mes statut à ses favoris
```
Comment (wydrop) [inline config/locales/simple_form.fr.yml]: It's not "mes statut" but "mes statuts"
  favourite: Envoyer un courriel lorsque quelqu’un ajoute mes statuts à ses favoris

File: config/locales/doorkeeper.fr.yml [hunk 2]
```diff
@@ -67,6 +67,7 @@ fr:
         application: Application
         created_at: Créé le
         date_format: "%Y-%m-%d %H:%M:%S"
```
Comment (wydrop) [inline config/locales/doorkeeper.fr.yml]: In french it's: Day Month Years /
         date_format: "%d-%m-%Y %H:%M:%S"

File: config/locales/fr.yml [hunk 3]
```diff
@@ -13,26 +34,58 @@ fr:
...
   auth:
     change_password: Changer de mot de passe
     didnt_get_confirmation: Vous n’avez pas reçu les consignes de confirmation ?
     forgot_password: Mode passe oublié ?
     login: Se connecter
+    logout: Se déconnecter
     register: S’inscrire
     resend_confirmation: Envoyer à nouveau les consignes de confirmation
     reset_password: Réinitialiser le mot de passe
-    set_new_password: Établir le nouveau mot de passe
```
Comment (wydrop) [inline config/locales/fr.yml]: set_new_password: Établir un nouveau mot de passe

File: config/locales/fr.yml [hunk 4]
```diff
@@ -48,8 +101,44 @@ fr:
...
+    acct: Entrez votre pseudo@instance depuis lequel vous voulez suivre ce⋅tte utilisateur⋅trice
+    missing_resource: L'URL de redirection n'a pas pu être trouvée
+    proceed: Continuez pour suivre
+    prompt: 'Vous allez suivre :'
   settings:
+    authorized_apps: Applications autorisées
+    back: Retour vers Mastodon
     edit_profile: Modifier le profil
+    export: Export de données
+    import: Import
```
Comment (wydrop) [inline config/locales/fr.yml]: I
    import: Importer des données

File: config/locales/fr.yml (hunk 3 above)
Comment (eramdam) [inline config/locales/fr.yml]: I'm not sure, "Etablir" doesn't sound very usual in French, I never saw it in French UIs anyway.

**Pull request merged**